engine/execution/computation/computer/uploader/model.go

```diff
@@ -0,0 +1,50 @@
+package uploader
+
+import (
+	"io"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/onflow/flow-go/engine/execution"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/mempool/entity"
+)
+
+type BlockData struct {
+	Block       *flow.Block
+	Collections []*entity.CompleteCollection
+	TxResults   []*flow.TransactionResult
+	Events      []*flow.Event
+	TrieUpdates []*ledger.TrieUpdate
+}
+
+func ComputationResultToBlockData(computationResult *execution.ComputationResult) *BlockData {
+
+	txResults := make([]*flow.TransactionResult, len(computationResult.TransactionResults))
+	for i := 0; i < len(computationResult.TransactionResults); i++ {
+		txResults[i] = &computationResult.TransactionResults[i]
+	}
+
+	events := make([]*flow.Event, 0)
+	for _, eventsList := range computationResult.Events {
+		for i := 0; i < len(eventsList); i++ {
+			events = append(events, &eventsList[i])
+		}
```

Comment:
Yep, go gotcha, when iteration variable is reused and it has the same address all the time.

Reply:
No.
`event` in most inner loop would be of type `Event` and we want `*Event`. 
If you do `events = append(events, &event)` you would append the same element multiple times, since `event` is reused in the loop and only initialized once

Reply:
Ah yes, so this instead

```go
for _, eventsList := range computationResult.Events {
    for _, event := range eventsList {
      e := &event
	    events = append(events, e)
    }
}
```